src/main/java/io/kestra/plugin/azure/cli/AzCLI.java

```diff
@@ -0,0 +1,154 @@
+package io.kestra.plugin.azure.cli;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
+import io.kestra.plugin.scripts.exec.scripts.models.RunnerType;
+import io.kestra.plugin.scripts.exec.scripts.models.ScriptOutput;
+import io.kestra.plugin.scripts.exec.scripts.runners.CommandsWrapper;
+import io.kestra.plugin.scripts.exec.scripts.services.ScriptService;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "Execute az commands."
+)
+@Plugin(
+        examples = {
+                @Example(
+                        title = "List ActiveDirectory users for a tenant using a user account",
+                        code = {
+                                "username: \"<user>\"",
+                                "password: \"secret('az-password')\"",
+                                "tenant: \"<tenant-id>\"",
+                                "commands:",
+                                "  - az ad user list"
+                        }
+                ),
+                @Example(
+                        title = "List all tenant's VMs successfully provisioned using a service principal authentication",
+                        code = {
+                                "username: \"<app-id>\"",
+                                "password: \"secret('az-sp-pass-or-cert')\"",
+                                "tenant: \"<tenant-id>\"",
+                                "servicePrincipal: true",
+                                "commands:",
+                                "  - az vm list --query \"[?provisioningState=='Succeeded']\""
+                        }
+                ),
+                @Example(
+                        title = "Command without authentication",
+                        code = {
+                                "commands:",
+                                "  - az --help"
+                        }
+                )
+        }
+)
+public class AzCLI extends Task implements RunnableTask<ScriptOutput> {
+    @Schema(
+            title = "The commands to run."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @NotEmpty
+    private List<String> commands;
+
+    @Schema(
+            title = "Account username."
```

Comment:
Explain here or in the task description that if a username is set, the task will emit an `as login` command prior to the commands.